File: guides to algorithms/Guia07/Python/ArvorePesquisaBinaria/Pilha.py
from io import StringIO

class Pilha:

    n = 0
    top = -1
    S = []    

    def __init__(self, n):
        self.n = n
        self.top = -1
        self.S = [0] * n

    def stack_empty(self):
        if self.top == -1:
            return True
        else:
            return False

    def push(self, item):
        self.top = self.top + 1
        self.S[self.top] = item

    def pop(self):
        if self.stack_empty is False:
            print("underflow")
            return 0
        else:
            self.top = self.top - 1
            return self.S[self.top+1]

    def size(self):
        return len(self.S)

    def toString(self):
        txt = StringIO()
        for k in self.S:
            txt.write("{}".format(k))
            txt.write(" ")
        return txt.getvalue()

File: guides to algorithms/Guia08/Java/ArvoreRedBlack/Pilha.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package arvoresredblack;

public class Pilha {
    int n = 0;
    int top = -1;
    int[] S = null;
    public boolean stack_empty(){
        if(this.top == -1) return true;
        else return false;
    }
    
    public void push(int x){
        this.top = this.top + 1;
        this.S[this.top] = x;
    }
        
    public int pop(){
        if(this.stack_empty()){
            System.out.println("underflow");
            return 0;
        } else{
            this.top = this.top - 1;
            return this.S[this.top+1];
        }
    }
    
    public Pilha(int n){
        this.n = n;
        this.top = -1;
        this.S = new int[this.n];
    }
    
    @Override public String toString(){
        StringBuilder res = new StringBuilder();
        for(int x: this.S){
            res.append(x);
            res.append(" ");
        }
        return res.toString();
    }
}
